src/v3/src/components/Widget/index.tsx

```diff
@@ -231,7 +231,7 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
   }, [pollingTransaction]); // only watch on pollingTransaction changes
 
   useEffect(() => {
-    if (!idxTransaction) {
+    if (!idxTransaction || isClientTransaction) {
```

Comment:
Compared to v2, afterRender should only be called once the page successfully navigates (should not trigger when there was an error that prevented the page from navigating away/to the next step).